akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala

```diff
@@ -238,6 +239,18 @@ trait BasicDirectives {
   def extractMaterializer: Directive1[Materializer] = BasicDirectives._extractMaterializer
 
   /**
+   * Extracts the [[akka.actor.ActorSystem]] from the [[akka.http.scaladsl.server.RequestContext]].
```

Comment:
Please add that `Extracts the [[akka.actor.ActorSystem]] if the available Materializer is an [[ActorMaterializer]]. Otherwise throws an exception as it won't be able to extract the system from arbitrary materializers.`